submissions/ses3332002/movie-seat-booking-js/script.js

```diff
@@ -0,0 +1,163 @@
+
+let selectedSeats = [];
+
+const seatsForm = document.forms.auditorium__selection;
+let formSubmitButton;
+
+const auditorium = document.querySelector('.auditorium');
+const preOrder = document.querySelector('.pre_order');
+const ticketsList = preOrder.querySelector('.pre_order__list');
+const orderConfirmationWindow = document.querySelector('.order_confirmation');
+const orderConfirmationText = orderConfirmationWindow.querySelector('.order_confirmation__text');
+const confirmButton = orderConfirmationWindow.querySelector('.order_confirmation__button-confirm');
+const cancelButton = orderConfirmationWindow.querySelector('.order_confirmation__button-cancel');
+
+function prepareRow(rowData, rowIndex) {
+  let rowString = '';
+  rowData.forEach((item, i) => {
+    switch (item) {
+      case 0:
+        rowString += `<input type="checkbox" id="seat${`${rowIndex + 1}_${i + 1}`}"/><label for="seat${rowIndex + 1}_${
+          i + 1
+        }" class="seat" data-price="${regularSeatPrice}"></label>`;
+        break;
+      case 1:
+        rowString += `<input type="checkbox" id="seat${rowIndex + 1}_${i + 1}" disabled /><label for="seat${
+          rowIndex + 1
+        }_${i + 1}" class="seat"></label>`;
+        break;
+      case 2:
+        rowString += `<input type="checkbox" id="seat${rowIndex + 1}_${i + 1}" disabled checked /><label for="seat${
+          rowIndex + 1
+        }_${i + 1}" class="seat"></label>`;
+        break;
+      case 3:
+        rowString += `<input type="checkbox" id="seat${rowIndex + 1}_${i + 1}" data-type="vip"><label for="seat${
+          rowIndex + 1
+        }_${i + 1}" class="seat" data-price="${vipSeatPrice}" /></label>`;
+        break;
+      case 4:
+        rowString += `<input type="checkbox" id="seat${rowIndex + 1}_${
+          i + 1
+        }" disabled checked data-type="vip"/><label for="seat${rowIndex + 1}_${i + 1}" class="seat"/></label>`;
+    }
+  });
+  return rowString;
+}
+
+function initAuditorium() {
+  let renderString = '';
+  seats.forEach((item, i) => {
+    renderString += `<div class="auditorium__row">${prepareRow(item, i)}</div>`;
+  });
+  renderString += '<input type="submit" class="seats_submit" value="Заказать">';
+  seatsForm.innerHTML = renderString;
+  formSubmitButton = seatsForm.querySelector('.seats_submit');
+  seatsForm.addEventListener('submit', submitSelectedTickets);
+  seatsForm.addEventListener('change', calculatePreOrder);
+}
+
+function prepareTicketsList() {
+  let renderString = '';
+  selectedSeats.forEach((item) => {
+    renderString += `<li class="pre_order__item">ряд ${item[0]}, место ${item[1]}</li>`;
+  });
+  ticketsList.innerHTML = renderString;
+}
+
+function calculatePreOrder() {
+  const checkedNodeList = seatsForm.querySelectorAll('input[type=checkbox]:checked:not(:disabled)');
+  if (checkedNodeList.length > 0) {
+    selectedSeats = [...checkedNodeList].map(({ id, dataset }) => [
+      parseInt(id.slice(4), 10),
```

Comment:
magic numbers?
Mabey better set `id` without letters?

Reply:
I've remade it